dashboard/client/src/pages/metrics/Metrics.tsx

```diff
@@ -0,0 +1,187 @@
+import {
+  Button,
+  createStyles,
+  makeStyles,
+  MenuItem,
+  Paper,
+  TextField,
+} from "@material-ui/core";
+import { Alert } from "@material-ui/lab";
+import React, { useContext, useEffect, useState } from "react";
+
+import { GlobalContext } from "../../App";
+
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    root: {},
+    grafanaEmbedsContainer: {
+      marginTop: theme.spacing(1),
+      marginLeft: theme.spacing(1),
+      display: "flex",
+      flexDirection: "row",
+      flexWrap: "wrap",
+    },
+    grafanaEmbed: {
+      margin: theme.spacing(1),
+    },
+    topBar: {
+      position: "sticky",
+      width: "100%",
+      display: "flex",
+      flexDirection: "row",
+      alignItems: "center",
+      justifyContent: "flex-end",
+      padding: theme.spacing(1),
+    },
+    timeRangeButton: {
+      marginLeft: theme.spacing(2),
+    },
+  }),
+);
+
+enum TimeRangeOptions {
+  FIVE_MINS = "5 minutes",
+  THIRTY_MINS = "30 minutes",
+  ONE_HOUR = "1 hour",
+  THREE_HOURS = "3 hours",
+  SIX_HOURS = "6 hours",
+  TWELVE_HOURS = "12 hours",
+  ONE_DAY = "1 day",
+  TWO_DAYS = "2 days",
+  SEVEN_DAYS = "7 days",
+}
+
+const TIME_RANGE_DURATIONS_MS: Record<TimeRangeOptions, number> = {
+  [TimeRangeOptions.FIVE_MINS]: 1000 * 60 * 5,
+  [TimeRangeOptions.THIRTY_MINS]: 1000 * 60 * 30,
+  [TimeRangeOptions.ONE_HOUR]: 1000 * 60 * 60 * 1,
+  [TimeRangeOptions.THREE_HOURS]: 1000 * 60 * 60 * 3,
+  [TimeRangeOptions.SIX_HOURS]: 1000 * 60 * 60 * 6,
+  [TimeRangeOptions.TWELVE_HOURS]: 1000 * 60 * 60 * 12,
+  [TimeRangeOptions.ONE_DAY]: 1000 * 60 * 60 * 24 * 1,
+  [TimeRangeOptions.TWO_DAYS]: 1000 * 60 * 60 * 24 * 2,
+  [TimeRangeOptions.SEVEN_DAYS]: 1000 * 60 * 60 * 24 * 7,
+};
+
+const METRICS_CONFIG = [
+  {
+    title: "Instance count",
+    path: "/d-solo/rayDefaultDashboard/default-dashboard?orgId=1&theme=light&panelId=24",
```

Comment:
Btw, just OOC, is this difficult to make it automatically find all dashboard from Grafana?

Reply:
Yes. It's not possible today. Instead, grafana supports embedding the entire dashboard page but it looks very ugly.